Mono.Debugging/Mono.Debugging.Client/Assembly.cs

```diff
@@ -0,0 +1,122 @@
+//
+// Assembly.cs
+//
+// Author:
+//       Jonathan Chang <t-jochang@microsoft.com>
+//
+// Copyright (c) 2022 
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+using System;
+namespace Mono.Debugging.Client
+{
+	public class Assembly
+	{
+		string name;
+		string path;
+		bool optimized;
+		bool usercode;
+		string symbolStatus;
+		string symbolFile;
+		int order;
+		string version;
+		string timestamp;
+		string address;
+		string process;
+		string appdomain;
+
+		public Assembly (string name, string path, bool optimized, bool userCode, string symbolStatus, string symbolFile, int? order, string version, string timestamp, string address, string process, string appdomain)
+		{
+			Name = name;
+			Path = path;
+			Optimized = optimized;
+			SymbolStatus = symbolStatus;
+			SymbolFile = symbolFile;
+			if (order == null) {
+				Order = -1;
+			} else {
+				Order = (int)order;
+			}
+
+			TimeStamp = timestamp;
+			Address = address;
+			Process = process;
+			AppDomain = appdomain;
+			Version = version;
+			usercode = userCode;
+		}
+		public Assembly (string path)
+		{
+			Path = path;
+		}
+		public string Name {
```

Comment:
Minor nit. Then all of these don't really need to have backing fields, nor implementations, since the compiler will auto-generate the same code.

```suggestion
		public string Name { get; protected set; }		
```